src/main/java/com/bootcamp/demo/restapi/BookingsController.java

```diff
@@ -0,0 +1,174 @@
+        }
+    }
+
+    @GetMapping("/showAllBookings")
+    @ResponseBody
+    public ResponseEntity<LinkedHashSet<Booking>> getAllBookings() {
+        LinkedHashSet<Booking> bookings = new LinkedHashSet<>();
+        try {
+            bookings = bookingService.getAllBookings();
+            return new ResponseEntity<>(bookings, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.info("Unfortunately,an error happened while trying to get all the bookings.");
+
+            return new ResponseEntity<>(bookings, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PostMapping ("/bookScooter")
+    @ResponseBody
+    public void bookScooter(final Double lat, final Double longitude) throws ExecutionException, InterruptedException {
+        bookAScooter(lat, longitude);
+
+    }
+
+    @DeleteMapping("/delete")
+    public ResponseEntity<Object> deleteBooking(@RequestParam final String id) {
+        try {
+            LOGGER.info("DELETE BOOKING - API endpoint invoked");
+            bookingService.deleteBooking(id);
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            LOGGER.info("Unfortunately,an error happened when trying to delete a booking.");
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PutMapping("/update")
+    public ResponseEntity<Object> updateBooking(@RequestParam final String id, @RequestParam("end")
+            String endDate, PaymentStatus status) {
+        try {
+            LOGGER.info("UPDATE BOOKING - API endpoint invoked");
+
+            Booking booking = bookingService.getBookingByID(id);
+            Scooter scooter = scooterService.findScooterById(booking.getSerialNumber());
+
+            if (status == PaymentStatus.SUCCESS) {
+                endBookingAndUpdate(booking, scooter, endDate);
+            }
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            LOGGER.info("Unfortunately,an error happened when trying to update the booking.");
+
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    public double endBookingAndUpdate(Booking booking, Scooter scooter, String endDate) throws ExecutionException, InterruptedException {
+        long bookingDuration = 0;
+
+        String startDate = booking.getStartDate();
+        //Date endDate = booking.getEndDate();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+        LocalDateTime dateTimeStart = LocalDateTime.parse(startDate, formatter);
+        LocalDateTime dateTimeEnd = LocalDateTime.parse(endDate, formatter);
+        if (dateTimeEnd.isAfter(dateTimeStart)) {
+            bookingDuration = dateTimeEnd.getMinute() - dateTimeStart.getMinute();
```

Comment:
Will this computation work as expected for all scenarios i.e. dateTimeEnd minute is smaller than dateTimeStart minute?
You could find other methods to substract dates and get the result in minutes (hint: Duration class)